Mathlib/CategoryTheory/Triangulated/HomologicalFunctor.lean

```diff
@@ -61,6 +57,17 @@ namespace Functor
 
 variable (F : C ⥤ A)
 
+/-- The kernel of a homological functor `F : C ⥤ A` is the strictly full
+triangulated subcategory consisting of objects `X` such that
+for all `n : ℤ`, `F.obj (X⟦n⟧)` is zero. -/
+def homologicalKernel : ObjectProperty C :=
+  fun X ↦ ∀ (n : ℤ), IsZero (F.obj (X⟦n⟧))
+
+lemma mem_homologicalKernel_iff [F.ShiftSequence ℤ] (X : C) :
+    F.homologicalKernel X ↔ ∀ (n : ℤ), IsZero ((F.shift n).obj X) := by
+  simp only [← fun (n : ℤ) => Iso.isZero_iff ((F.isoShift n).app X)]
```

Comment:
This is the first time that I see a `simp […]` like that! :-)
The final `rfl` seems to be needed to rewrite `(shiftFunctor C n ⋙ F).obj X` to `F.obj (X⟦n⟧)` of the definition of `homologicalKernel`. I wonder whether this justifies a lemma.